src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -0,0 +1,289 @@
+            **kwargs,
+        )
+    except SandboxExecutionError as sdbx_err:
+        cc.warning(
+            f"Could not fetch Snowpark objects from {py_file} due to {sdbx_err}, continuing execution for the rest of the python files."
+        )
+        return None
+    except Exception as err:
+        cc.warning(
+            f"Could not fetch Snowpark objects from {py_file} due to {err}, continuing execution for the rest of the python files."
+        )
+        return None
+
+    if completed_process.returncode != 0:
+        cc.warning(
+            f"Could not fetch Snowpark objects from {py_file} due to the following error:\n {completed_process.stderr}"
+        )
+        cc.warning("Continuing execution for the rest of the python files.")
+        return None
+
+    try:
+        return json.loads(completed_process.stdout)
+    except Exception as exc:
+        cc.warning(
+            f"Could not load JSON into python due to the following exception: {exc}"
+        )
+        cc.warning(f"Continuing execution for the rest of the python files.")
+        return None
+
+
+def _add_py_file_dest_to_dict(
+    dest_path: Path,
+    py_file: Path,
+    src_py_file_to_dest_py_file_map: Dict[Path, Path],
+    deploy_root: Path,
+):
+    dest_file = dest_path / py_file.name
+    if (
+        dest_file.exists()
+    ):  # Should already exist since bundle is called before processing begins
+        src_py_file_to_dest_py_file_map[py_file] = dest_file
+    else:
+        cc.warning(f"{dest_path} does not exist in {deploy_root}.")
+
+
+class SnowparkAnnotationProcessor(ArtifactProcessor):
+    """
+    Built-in Processor to discover Snowpark-annotated objects in a given set of python files,
+    and generate SQL code for creation of extension functions based on those discovered objects.
+    """
+
+    def __init__(
+        self,
+        project_definition: NativeApp,
+        project_root: Path,
+        deploy_root: Path,
+    ):
+        super().__init__(
+            project_definition=project_definition,
+            project_root=project_root,
+            deploy_root=deploy_root,
+        )
+        self.project_definition = project_definition
+        self.project_root = project_root
+        self.deploy_root = deploy_root
+
+    def process(
+        self,
+        artifact_to_process: PathMapping,
+        processor_mapping: Optional[ProcessorMapping],
+    ) -> Dict[Path, Optional[Any]]:
+        """
+        Intended to be the main method which can perform all relevant processing, and/or write to a target file, which depends on the type of processor.
+        For SnowparkAnnotationProcessor, the target file is the setup script.
+        """
+
+        kwargs = (
+            _determine_virtual_env(processor_mapping)
+            if processor_mapping is not None
+            else {}
+        )
+
+        # 1. Get all src.py -> dest.py mapping
+        src_py_file_to_dest_py_file_map = self.get_src_py_file_to_dest_py_file_map(
```

Comment:
note: I'm working on an alternative to this, but let's go with this implementation for now so we can keep making progress. This code will be replaced soon.

Reply:
I've added a TODO here noting the impending change.